### DISCUSSION TITLE
Arbitrary inserted_at in the example realtime chat app?

### DISCUSSION THREAD

**Discussion by timas130** [supabase/supabase#6049] [Questions] [answered]
So I was looking at [this schema](https://github.com/supabase/supabase/blob/master/examples/nextjs-slack-clone/full-schema.sql) and noticed that it does not check the `inserted_at` column in `public.messages`. I tried to use it in the hosted example, and it is possible to pass arbitrary values into this column.

I ran into the same problem with my project. Is it intentional and how do I avoid it?

**Comment by GaryAustin1** [comment, marked as answer]
I use the moddatetime extension (available in extensions tab).
Basically you put a before trigger on the table and call the function with your updated at column.  User input is ignored then...
Here is their example:
```
CREATE TABLE mdt (
	id		int4,
	idesc		text,
	moddate timestamp DEFAULT CURRENT_TIMESTAMP NOT NULL
);

CREATE TRIGGER mdt_moddatetime
	BEFORE UPDATE ON mdt
	FOR EACH ROW
	EXECUTE PROCEDURE moddatetime (moddate);
```

**Reply by timas130** [reply]
What about blocking changes to a column? I have a `created_at` column, I don't want users to change it.

**Reply by GaryAustin1** [reply]
Ah, I read your question a bit wrong...  I use a before trigger and then call a function that does:
```
begin
new.created_at = old.created_at;
return new;
end;
```
There probably is a way to do it with column security, but I've not looked into that.

**Reply by timas130** [reply]
Thanks a lot. That's what I was thinking about, but thought that there should be a better way as well.

**Comment by GaryAustin1** [comment]
sorry,
